### ISSUE TITLE
Error 404 downloading package

### ISSUE THREAD

**Issue by J-Hendy** [retorquere/zotero-deb#59]
Hi,

I'm a bit rusty with adding package repos but followed the installing Zotero steps and had the following 404 errors when downloading the package:

```
sudo apt install zotero
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  zotero
0 to upgrade, 1 to newly install, 0 to remove and 0 not to upgrade.
Need to get 64.1 MB of archives.
After this operation, 0 B of additional disk space will be used.
Err:1 https://downloads.sourceforge.net/project/zotero-deb ./ zotero 5.0.96.3-3
  404  Not Found [IP: 204.68.111.105 443]
E: Failed to fetch https://downloads.sourceforge.net/project/zotero-deb/zotero-better-bibtex/zotero_5.0.96.3-3_amd64.deb  404  Not Found [IP: 204.68.111.105 443]
```

Regards.

**Comment by retorquere** [comment]
It should be back up. I'm in the process of setting up something more stable. sourceforge has turned out to be really cantankerous.

**Comment by mateMathieu** [comment]
All our support for this!
Let us know here when you have a chance to complete it.
Thank you so much.

**Comment by retorquere** [comment]
I will let you know, I'm in the process of setting up the new mirror.

**Comment by retorquere** [comment]
> rsync over ssh would be perfect. I guess I would make you a user account on my server and you add the rsync command to GitHub actions with your password stored as a secret?

That's how I have it set up for sourceforge right now. I hope Ruben won't take affront, but I'm eager to get away from sourceforge and backblaze.

> Note: My mirror is accessible through two hostnames: <mirror.mwt.me> and <mirror-cf.mwt.me>. The second uses free Cloudflare as you described.

I'd just point my existing cloudflare DNS at either of those.

**Comment by mwt** [comment]
That works for me. You can just send me your pubkey to add to `authorized_hosts`. I'll make a retorquere user for you and you can rsync via retorquere@mirror-rs.mwt.me

You should probably point Cloudflare at mirror-rs.mwt.me. I'll add your hostname to nginx.

**Comment by retorquere** [comment]
```
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC/z3vcoTsQ1g9TAsDz7pBPLqJtJqcpfpCkk1XYAgEXR8Mkh3N9xbwIBLCDup9dOh40brBLPZgR8JsmuehOXHSx9XfuQlZKlK+Q6sOf9WEC6szrQTkI5puwSyjc/KCmbb/T8MaSUAmenS1ilTzphQvb0YqjyXakNVb4B2kv5iDiT9IzAiOwIAQZMKGoXtKKSqvRZPj6WsD4Cx47c4oQOFsiIStuSm0IGxxgvbaFqGf8hQAW4XYceskjs3Gpa2H7quYw4xUDsNQ3PUtw6Ab/FAcYQc/Im8pN77EIOROIKRYR59EYqVbgBJEHJP0tijbiZQTwF27wkfHx5R2GkHgNRk4zRhCt/eghDf7nLJg8fwn0bBBuX1orm+tUhGMYOu0rXcliXvmq5n7s6EaeShsQ+Z6pd0YbvfeQklLVmorQRQzr3a+Bbho9TEqQ8NlWgSqshSe8gxc47MGCnqLIvUnxjspvVuwxtq53K3aJXVQ+siXzFKFySjEvr04lbH9PEo3QU7TGCDolv6YZp1AncexKXgyQYYnQxmCM6lN40FQbI61K0FxV0Y3C3Dl2Y9PN4fgT5R1wM4cZJn5iUU8dfQjyYhnSWt8NBUwREDl51p7lakLs0oJzySLZG6fDPToBjfrk4rDXsRFRi9UBMcP3MWnyn40DW7RGfGMndaC4UUoMv/Dj3w== emiliano.heyns@iris-advies.com
```
Can you add these two hostnames?

- apt.retorque.re (full repo URL: https://apt.retorque.re/zotero-apt)
- zotero.retorque.re (full repo URL: https://zotero.retorque.re/apt-package-archive)

I want to add rpm packages too at some point so I want to gradually migrate everything to the latter hostname.

**Comment by mwt** [comment]
I added the user. You should be able to push with something like
```
rsync -av --delete $SOURCE retorquere@mirror-rs.mwt.me:/
```
I setup nginx on those two hostnames to show the same site as [mirror.mwt.me](https://mirror.mwt.me) but without the redirect to my website. We can adjust this once there are some synced files to display.

**Comment by retorquere** [comment]
Do these map to the same actual location? In that case I'll just use the latter -- the final part (`zotero-apt`/`apt-package-archive`) is in the repo content... which I now realize is because the Zotero crew asked me to do this... which complicates *everything*. I may need to rethink this.

**Comment by retorquere** [comment]
Yeah, removing the final part from the package contents makes total sense. I'm trying to get the rebuild with it stable tonight, then tomorrow without, and then we don't need to round-robin anything. Any mirror under any URL would work.

**Comment by mwt** [comment]
They both map to the same location using symlinks (which is `/` in rsync). I'm not sure what you mean by the final part being in the content.

It might be smarter to work this out such that you rsync to `/apt-package-archive` and then I serve the root using the webserver. Then, if you decide to sync rpm packages to `/dnf-archive` or something, it'll work without having to change anything.

**Comment by retorquere** [comment]
I mean that the `Packages` files has `apt-package-archive` in it for the one and `zotero-apt` for the other, and that the repo won't work for the mismatch. It doesn't actually have to have this in the `Packages` file, it didn't use to be there. I can revert that but need to get to a stable publish first.

**Comment by mwt** [comment]
Got it. That does sound like a pain. I just ripped these files with a simple:

```sh
wget -m https://apt.retorque.re/file/zotero-apt/
```
So, only `/zotero-apt` would be correct at the moment.

**Comment by retorquere** [comment]
The repos have been updated. What paths should I rsync to?

**Comment by mwt** [comment]
Let's do `retorquere@mirror-rs.mwt.me:/apt-package-archive`. That should be more flexible. That way you can sync to any `/path` and serve it as `https://zotero.retorque.re/path`.

**Comment by mwt** [comment]
It tried to use password auth on the server for some reason. Perhaps the SF cert install step is only setting it to be used by that one host.

I'll troubleshoot by trying to rsync something simpler through gh actions.
